src/main/java/org/candlepin/subscriptions/tally/admin/OrgIdToTallySnapshotMigration.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright Red Hat, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * Red Hat trademarks are not licensed under GPLv3. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.subscriptions.tally.admin;
+
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.MeterRegistry;
+import java.sql.Types;
+import java.util.ArrayList;
+import java.util.List;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.SqlRowSetResultSetExtractor;
+import org.springframework.jdbc.support.rowset.SqlRowSet;
+import org.springframework.util.StringUtils;
+
+@Slf4j
+public class OrgIdToTallySnapshotMigration extends DataMigration {
+
+  public static final SqlRowSetResultSetExtractor SQL_ROW_SET_RESULT_SET_EXTRACTOR =
+      new SqlRowSetResultSetExtractor();
+
+  public static final String UPDATE_SQL =
+      "update tally_snapshots set org_id=? where id =? and org_id=''";
```

Comment:
This doesn't work with null orgIds; only empty string orgIds.  If you insert the records below, the migration does not work.  It will find the records thanks to the `!StringUtils.hasText(tallyOrg)` check but the insert won't do anything because both condition aren't true.

```sql
INSERT INTO public.tally_snapshots (id, product_id, account_number, granularity, org_id, snapshot_date,
unit_of_measure, sla, usage, billing_provider, billing_account_id) VALUES ('c5310000-46bd-4991-94f8-dccadab62245',
'rhosak', 'account123', 'DAILY', null, '2021-10-31 00:00:00.000000 +00:00', null, 'Premium', '_ANY', 'red hat',
'dummyId');
INSERT INTO public.tally_snapshots (id, product_id, account_number, granularity, org_id, snapshot_date,
unit_of_measure, sla, usage, billing_provider, billing_account_id) VALUES ('8d810000-0c6a-432a-a4b4-a96a43186fd4',
'rhosak', 'account123', 'DAILY', null, '2021-10-31 00:00:00.000000 +00:00', null, '_ANY', 'Production', 'red hat',
'dummyId');
```

We know the orgIds are null and not empty string using Gabi:

```bash
% http -b https://gabi-subscription-watch-stage.apps.crcs02ue1.urby.p1.openshiftapps.com/query query="select count(id) from tally_snapshots where org_id =''" 'Authorization:Bearer $TOKEN'
{
    "error": "",
    "result": [
        [
            "count"
        ],
        [
            "0"
        ]
    ]
}

% http -b https://gabi-subscription-watch-stage.apps.crcs02ue1.urby.p1.openshiftapps.com/query query="select count(id) from tally_snapshots where org_id is null" 'Authorization:Bearer $TOKEN'
{
    "error": "",
    "result": [
        [
            "count"
        ],
        [
            "3248"
        ]
    ]
}
```

I would just remove the `and org_id=''` clause.  We already know we want to update that record; there's no reason to second-guess it here.